apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/ThrowableTransformer.java

```diff
@@ -30,10 +32,12 @@
     public String convert2String(Throwable throwable, final int maxLength) {
         final StringBuilder stackMessage = new StringBuilder();
         Throwable causeException = throwable;
-        while (causeException != null) {
+
+        int depth = CAUSE_EXCEPTION_DEPTH;
+        while (causeException != null || depth == 0) {
             stackMessage.append(printExceptionInfo(causeException));
 
-            boolean overMaxLength = printStackElement(throwable.getStackTrace(), new AppendListener() {
+            boolean isLookDeeper = printStackElement(causeException.getStackTrace(), new AppendListener() {
```

Comment:
@Arugal @SummerOfServenteen I think there are two places originally wrong, which also trigger #2926

Thank @Arugal to catch this.

Reply:
maybe depth != 0

Reply:
maybe ( causeException != null  && depth != 0)

Reply:
Yes. Done.